src/field.cpp

```diff
@@ -2071,6 +2096,31 @@ void map::player_in_field( player &u )
             }
             break;
 
+        case fd_fungicidal_gas:
+            // Fungicidal gas makes you cough.
+            // Thick fungicidal gas has a chance to poison you.
+            {
+                bool inhaled = false;
+                if( cur->getFieldDensity() == 3 ) {
+                    inhaled = u.add_env_effect("poison", bp_mouth, 5, 30);
+                } else if( cur->getFieldDensity() == 2 ) {
+                    inhaled = u.add_env_effect("smoke", bp_mouth, 2, 7);
+                } else if( u.has_trait("THRESH_MYCUS") || u.has_trait("THRESH_MARLOSS" ) ) {
```

Comment:
Both of those `else if`s are wrong:
* Thick gas will not cause coughing, only poison. In many cases coughing can actually be more dangerous than poison
* Mycus mutant will only get the mycus effect if the density == 1
* In the mycus block: you don't need to compare density to any of 1, 2 or 3. It can never be lower than 1 or higher than 3
* Density 1 should do something too. Maybe just a weak (5 turns?) poison effect?
* Mycus mutants will take the damage even when in full hazmat